### PR TITLE
ProgData rework progress as it was before

### PR DESCRIPTION
**Pull request by julianvu**
Progress on the ProgData rework was made before the rebasing issue. This pull request involves the work in reapplying that previous work onto the fixed develop branch.

### CODE REVIEW

File: src/ModelBasedPrognoser.cpp [hunk 1]
```diff
@@ -117,13 +120,12 @@ namespace PCOE {
 
         // Create progdata
         results.setUncertainty(UType::Samples); // @todo(MD): do not force samples representation
-        for (std::string& event : model->events) {
+        for (std::string & event : model->events) {
```
Comment (jason-watkins) [inline src/ModelBasedPrognoser.cpp]: Incorrect formatting. Run clang-format

File: src/MonteCarloPredictor.cpp [hunk 2]
```diff
@@ -96,13 +94,30 @@ namespace PCOE {
...
+        predictionSysTraj.setNPoints(numSamples);
+
         auto stateTimestamp = getLowestTimestamp(state);
 
-// Create a random number generator if operating sequentially
-#ifndef USING_OPENMP
-        static std::random_device rDevice;
-        static std::mt19937 generator(rDevice());
-#endif
+        // Create a random number generator if operating sequentially
```
Comment (jason-watkins) [inline src/MonteCarloPredictor.cpp]: clang-format

File: src/ProgEvent.cpp [hunk 3]
```diff
@@ -41,8 +41,7 @@ namespace PCOE {
     // *------------------------*
 
     ProgEvent::ProgEvent() : ProgMeta(), nSamples(0) {
-        probMatrix.resize(1, NAN);
-        occurrenceMatrix.resize(1, std::vector<bool>(nSamples));
+
```
Comment (jason-watkins) [inline src/ProgEvent.cpp]: clang-format

File: src/RecorderCommunicator.cpp [hunk 4]
```diff
@@ -39,7 +39,7 @@
  *   @note      This class will look for the following optional configuration parameters:
  *                  saveFile    File to which the data will be saved (default RecordedMessages.csv")
  *
- *   @see        Communicator
+ *   @see        CommonCommunicator
```
Comment (jason-watkins) [inline src/RecorderCommunicator.cpp]: Should be Communicator

File: Test/gsapTests/DPointsTests.cpp [hunk 5]
```diff
@@ -51,9 +51,6 @@ void testPEventsInit() {
     // Set Num Times
     test.setNTimes(5);
     Assert::AreEqual(5, test.getNTimes());
-    Assert::AreEqual(5, test["Test"].getNumTimes());
-    test.addNew("TestNew2", "Test Desc");
-    Assert::AreEqual(5, test["TestNew2"].getNumTimes());
```
[the remark this answers is not in the document]
Comment (julianvu) [reply Test/gsapTests/DPointsTests.cpp]: Occurrence matrices are involved in the getNumTimes() function.

File: inc/Predictor.h [hunk 6]
```diff
@@ -61,18 +70,18 @@ namespace PCOE {
         }
 
         /** @brief Set model pointer
-         *  @param value given model pointer
+         *  @param model given model pointer
          **/
-        virtual void setModel(PrognosticsModel* value) {
```
Comment (jason-watkins) [inline inc/Predictor.h]: I want to keep `value` here. Visual Studio doesn't like parameter names that shadow class members.

File: inc/MonteCarloPredictor.h [hunk 7]
```diff
@@ -50,7 +50,7 @@ namespace PCOE {
          *  @param    state state of system at time of prediction
          *  @param  data ProgData object, in which prediction results \re stored
          **/
-        void predict(const double tP, const std::vector<UData>& state, ProgData& data);
+        PCOE::Prediction predict(const double tP, const std::vector<UData>& state, ProgData& data);
```
Comment (jason-watkins) [inline inc/MonteCarloPredictor.h]: I don't think the namespace qualifier (`PCOE::`) is necessary here. Does anything fail to compile if you remove this?

File: src/MonteCarloPredictor.cpp [hunk 8]
```diff
@@ -84,9 +82,9 @@ namespace PCOE {
     }
 
     // Predict function
-    void MonteCarloPredictor::predict(const double time_s,
-                                      const std::vector<UData>& state,
-                                      ProgData& data) {
+    PCOE::Prediction MonteCarloPredictor::predict(const double time_s,
```
Comment (jason-watkins) [inline src/MonteCarloPredictor.cpp]: Same comment about namespace qualifier

File: src/MonteCarloPredictor.cpp [hunk 9]
```diff
@@ -61,8 +59,8 @@ namespace PCOE {
         log.WriteLine(LOG_INFO, MODULE_NAME, "MonteCarloPredictor created");
     }
 
-    void MonteCarloPredictor::setModel(PrognosticsModel* value) {
```
Comment (jason-watkins) [inline src/MonteCarloPredictor.cpp]: Same comment about param name.

File: inc/Predictor.h [hunk 10]
```diff
@@ -61,7 +70,7 @@ namespace PCOE {
         }
 
         /** @brief Set model pointer
-         *  @param value given model pointer
+         *  @param model given model pointer
```
Comment (jason-watkins) [inline inc/Predictor.h]: Forgot to change the doc comment name

**Pull request merged**